tests/functional/services/flows/test_list_runs.py

```diff
@@ -68,3 +69,37 @@ def test_list_runs_filter_flow_id(flows_client, pass_as_uuids):
     )
     for run in res_combined:
         assert run["flow_id"] in {str(flow_id_one), str(flow_id_two)}
+
+
+@pytest.mark.parametrize(
+    "filter_roles, expected_filter_roles",
+    [
+        # empty list
+        ([], []),
+        # list with empty string
+        ([""], [""]),
+        # None
+        (None, None),
+        # single role as string
+        ("foo", ["foo"]),
+        # single role as list
+        (["foo"], ["foo"]),
+        # multiple roles as comma-separated string
+        ("foo,bar", ["foo,bar"]),
+        # multiple roles as list
+        (["foo", "bar"], ["foo,bar"]),
+    ],
+)
+def test_list_runs_filter_roles(flows_client, filter_roles, expected_filter_roles):
+    load_response(flows_client.list_runs).metadata
+    res = flows_client.list_runs(filter_roles=filter_roles)
+    assert res.http_status == 200
+
+    req = get_last_request()
+    assert req.body is None
+    parsed_qs = urllib.parse.parse_qs(urllib.parse.urlparse(req.url).query)
+
+    if expected_filter_roles and any(role for role in expected_filter_roles):
```

Comment:
The above also lets us simplify this check:
```suggestion
    if expected_filter_roles:
```

The `any()` check was what initially drew my attention -- it seemed odd but passes when run.